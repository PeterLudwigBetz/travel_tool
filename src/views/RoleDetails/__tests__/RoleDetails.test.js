import React from 'react';
import sinon from 'sinon';
import { RoleDetails } from '..';

let wrapper;
const props = {
  travelTeamMembers: [
    {
      fullName: 'A user',
      centers: [
        {
          id: 12345,
          location: 'Lagos, Nigeria'
        }
      ]
    }
  ],
  closeModal: jest.fn(),
  fetchRoleUsers: sinon.spy(),
  updateUserCenter: jest.fn(),
  isLoading: false,
  getCurrentUserRole: ['Travel Administrator', 'Requester'],
  history: {
    push: sinon.spy()
  },
  openModal: jest.fn(),
  shouldOpen: false,
  modalType: 'new model',
  match: {
    params: {
      roleId: 335498
    }
  },
  roleName: 'Travel team member',
  fetchCenters: sinon.spy(),
  centers: [{}],
  putRoleData: sinon.spy()
};

const state = {
  headTitle: '',
  userDetail: ''
};

describe('<ROleDetails />', () => {
  beforeEach(() => {
    wrapper = shallow(<RoleDetails {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(`calls the fetchRoleUser prop method on
    componentDidMount`, () => {
    expect(props.fetchRoleUsers.called).toEqual(true);
    expect(props.fetchRoleUsers.calledWith(335498)).toEqual(true);
  });

  it(`calls renders the RoleDetails Table with
    the correct number of users`, () => {
    const roleDetailsTable = wrapper
      .find('WithLoading')
      .dive()
      .find('RoleDetailsTable')
      .dive();
    expect(roleDetailsTable.find('.table__rows').length).toEqual(1);
  });
});
