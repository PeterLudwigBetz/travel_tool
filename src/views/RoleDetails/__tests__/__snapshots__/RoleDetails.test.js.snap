// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ROleDetails /> renders correctly 1`] = `
<Fragment>
  <Modal
    closeDeleteCommentModal={null}
    closeModal={[Function]}
    customModalStyles=""
    customOverlayStyle=""
    modalBar={<div />}
    modalContentId=""
    modalId=""
    params=""
    requestId=""
    showOverlay={true}
    title="Add User"
    visibility="invisible"
    width="480px"
  >
    <NewUserRoleForm
      centers={
        Array [
          Object {},
        ]
      }
      closeModal={[Function]}
      errors=""
      fetchCenters={[Function]}
      getRoleData={[Function]}
      handleUpdateRole={[Function]}
      myTitle="Add User"
      role="Travel team member"
      updateUserCenter={[Function]}
      updatingRole={false}
      userDetail=""
    />
  </Modal>
  <div
    className="rp-role__header"
  >
    <div
      className="role-panel-header"
    >
      <PageHeader
        actionBtn="Add User"
        openModal={[Function]}
        title="Travel team members"
        titleClassName="title"
      />
    </div>
  </div>
  <div
    className="rp-table"
  >
    <WithLoading
      error=""
      handleEditCenter={[Function]}
      isLoading={false}
      roleName="Travel team member"
      travelTeamMembers={
        Array [
          Object {
            "centers": Array [
              Object {
                "id": 12345,
                "location": "Lagos, Nigeria",
              },
            ],
            "fullName": "A user",
          },
        ]
      }
    />
  </div>
</Fragment>
`;
